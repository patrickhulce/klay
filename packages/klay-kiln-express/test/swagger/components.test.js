const defaultModelContext = require('klay-core').defaultModelContext
const components = require('../../dist/swagger/components')
const Cache = require('../../dist/swagger/cache').SwaggerSchemaCache
const utils = require('../utils')

describe('lib/swagger/components.ts', () => {
  let model

  beforeEach(() => {
    model = utils.state().model
  })

  describe('#getSchema', () => {
    it('should build an object schema', () => {
      const schema = components.getSchema(model)
      expect(schema).toMatchSnapshot()
    })

    it('should build an array schema', () => {
      const arrayModel = defaultModelContext.array().children(model)
      const schema = components.getSchema(arrayModel)
      expect(schema).toMatchSnapshot()
    })

    it('should reference schema from cache', () => {
      const arrayModel = defaultModelContext.array().children(model)
      const cache = new Cache()
      const arraySchema = components.getSchema(arrayModel, cache, 'Users')
      expect(arraySchema).toHaveProperty('$ref', '#/definitions/Users')
      const schema = components.getSchema(model, cache)
      expect(schema).toHaveProperty('$ref', '#/definitions/UsersItem')
    })
  })
})
