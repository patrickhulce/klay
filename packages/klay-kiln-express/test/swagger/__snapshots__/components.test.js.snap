// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/swagger/components.ts #getSchema should build an array schema 1`] = `
Object {
  "items": Object {
    "properties": Object {
      "age": Object {
        "format": "int64",
        "type": "integer",
      },
      "createdAt": Object {
        "format": "date-time",
        "type": "string",
      },
      "email": Object {
        "maxLength": 250,
        "type": "string",
      },
      "firstName": Object {
        "maxLength": 100,
        "type": "string",
      },
      "id": Object {
        "type": "string",
      },
      "isAdmin": Object {
        "type": "boolean",
      },
      "lastName": Object {
        "maxLength": 100,
        "type": "string",
      },
      "metadata": Object {
        "properties": Object {},
        "type": "object",
      },
      "password": Object {
        "maxLength": 32,
        "type": "string",
      },
      "tracking": Object {
        "properties": Object {
          "lastVisit": Object {
            "format": "date-time",
            "type": "string",
          },
          "trackingId": Object {
            "type": "string",
          },
        },
        "required": Array [
          "trackingId",
        ],
        "type": "object",
      },
      "updatedAt": Object {
        "format": "date-time",
        "type": "string",
      },
    },
    "required": Array [
      "id",
      "age",
      "isAdmin",
      "email",
      "password",
      "firstName",
      "lastName",
      "tracking",
      "createdAt",
      "updatedAt",
    ],
    "type": "object",
  },
  "type": "array",
}
`;

exports[`lib/swagger/components.ts #getSchema should build an object schema 1`] = `
Object {
  "properties": Object {
    "age": Object {
      "format": "int64",
      "type": "integer",
    },
    "createdAt": Object {
      "format": "date-time",
      "type": "string",
    },
    "email": Object {
      "maxLength": 250,
      "type": "string",
    },
    "firstName": Object {
      "maxLength": 100,
      "type": "string",
    },
    "id": Object {
      "type": "string",
    },
    "isAdmin": Object {
      "type": "boolean",
    },
    "lastName": Object {
      "maxLength": 100,
      "type": "string",
    },
    "metadata": Object {
      "properties": Object {},
      "type": "object",
    },
    "password": Object {
      "maxLength": 32,
      "type": "string",
    },
    "tracking": Object {
      "properties": Object {
        "lastVisit": Object {
          "format": "date-time",
          "type": "string",
        },
        "trackingId": Object {
          "type": "string",
        },
      },
      "required": Array [
        "trackingId",
      ],
      "type": "object",
    },
    "updatedAt": Object {
      "format": "date-time",
      "type": "string",
    },
  },
  "required": Array [
    "id",
    "age",
    "isAdmin",
    "email",
    "password",
    "firstName",
    "lastName",
    "tracking",
    "createdAt",
    "updatedAt",
  ],
  "type": "object",
}
`;
